src/prefect/server/api/concurrency_limits.py

```diff
@@ -84,17 +198,44 @@ async def read_concurrency_limit_by_tag(
     The `active slots` field contains a list of TaskRun IDs currently using a
     concurrency slot for the specified tag.
     """
+    # V2-first by tag; fall back to V1
+    v2_name = f"tag:{tag}"
 
     async with db.session_context() as session:
-        model = await models.concurrency_limits.read_concurrency_limit_by_tag(
-            session=session, tag=tag
-        )
+        model = await cl_v2_models.read_concurrency_limit(session=session, name=v2_name)
 
     if not model:
-        raise HTTPException(
-            status.HTTP_404_NOT_FOUND, detail="Concurrency limit not found"
+        # Fall back to V1 read before returning 404
+        async with db.session_context() as session:
+            v1_model = await models.concurrency_limits.read_concurrency_limit_by_tag(
+                session=session, tag=tag
+            )
+        if not v1_model:
+            raise HTTPException(
+                status.HTTP_404_NOT_FOUND, detail="Concurrency limit not found"
+            )
+        return v1_model
```

Comment:
This logic flow is wacky. I think we should keep the v2 logic at the top and v1 logic after it for consistency.